scripts/jenkins/cloud/ansible/roles/list_packages/library/diff_yaml.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+#
+# (c) Copyright 2020 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+
+from __future__ import print_function
+
+from re import findall
+
+from ansible.module_utils.basic import AnsibleModule
+
+from deepdiff import DeepDiff
+
+import yaml
+
+DOCUMENTATION = '''
+---
+module: diff_yaml
+short_description: do a diff of yaml data
+description: |
+  Module will compare 2 given yaml files and return
+  structured diff. All is done on local machine
+author: SUSE Linux GmbH
+options:
+  file1:
+    description: Path to the file
+  file2:
+    description: Path to the file
+  output:
+    description: Name of file to write diff into
+'''
+
+EXAMPLES = '''
+- diff_yaml:
+    file1: path/to/yaml/file
+    file2: path/to/yaml/file
+    output: path/to/diff/file
+  register: _result
+'''
+
+
+def loadFiles(file1, file2):
+    with open(file1) as f:
+        firstFile = yaml.safe_load(f)
+    with open(file2) as f:
+        secondFile = yaml.safe_load(f)
+    return (firstFile, secondFile)
+
+def addToDict(dictItems, valueObj, keyObj, item, action):
+    packageName = keyObj[0]
+    #packageName = action if len(keyObj) == 1 else keyObj[0]
+    attribute = keyObj[0] if len(keyObj) == 1 else keyObj[1]
+    #try:
+    #    attribute = keyObj[1]
+    #except:
+    #    attribute = ke
+    attrValue = valueObj
+    if packageName in dictItems:
+#        dictItems[packageName][attribute] = attrValue
+        dictItems[action][packageName][attribute] = attrValue
+    else:
+        # if dict for packageName is empty initiate it and put first record
+#        dictItems[packageName] = {}
+#        dictItems[packageName][attribute] = attrValue
+        dictItems[action] = {}
+        dictItems[action][packageName] = {}
+        if action != "values_changed":
+            dictItems[action][packageName]
```

Comment:
This one is not needed, right? or do you plan to add some value there?

Reply:
Well values_changed has different output. You can see that here(from my local tests):
```
dictionary_item_removed:
  This_package_was_removed: {}
values_changed:
  Mesa-libglapi0:
    version:                                                                                                                                                                                                                     
      new_value: 18.0.2
      old_value: 17.0.2
```

Reply:
I know that (tested myself) but the dict item is already initialized to empty value on line 74. So this code could look like this:
```
if action == "values_changed":
    dictItems[action][packageName][attribute] = attrValue
```
without the other case which does exactly nothing :o)

Reply:
Hmm, probably leftover from some tests. I will remove that part.